syncstorage/db.go

```diff
@@ -887,12 +887,15 @@ func (d *DB) getBSOs(
 		orderBy = "ORDER BY Modified ASC "
 	}
 
-	limitStmt := "LIMIT ?"
-	values = append(values, limit)
-
-	if offset != 0 {
-		limitStmt += " OFFSET ?"
-		values = append(values, offset)
+	var limitStmt string
+	if limit > 0 {
+		limitStmt = "LIMIT ?"
+		values = append(values, limit)
+
+		if offset != 0 {
+			limitStmt += " OFFSET ?"
+			values = append(values, offset)
```

Comment:
The protocol permits you to specify `offset` without `limit`, e.g. you could say "give me the first 100 items" and then use a second request to say "give me the rest from this offset".  I don't believe any clients actually do this, and I can't think of a reason why they would, but it's permitted by the protocol.